src/ImageFitter/ImageFitter.cc

```diff
@@ -0,0 +1,231 @@
+        gsl_vector_set(_fit_params, i * 6 + 5, component.pa());
+    }
+    _fdf.p = p;
+}
+
+int ImageFitter::SolveSystem() {
+    gsl_multifit_nlinear_parameters fdf_params = gsl_multifit_nlinear_default_parameters();
+    const gsl_multifit_nlinear_type* T = gsl_multifit_nlinear_trust;
+    const double xtol = 1.0e-8;
+    const double gtol = 1.0e-8;
+    const double ftol = 1.0e-8;
+    const bool print_iter = false;
+    const size_t n = _fdf.n;
+    const size_t p = _fdf.p;
+    gsl_multifit_nlinear_workspace* work = gsl_multifit_nlinear_alloc(T, &fdf_params, n, p);
+    gsl_vector* f = gsl_multifit_nlinear_residual(work);
+    gsl_vector* y = gsl_multifit_nlinear_position(work);
+    gsl_matrix* covar = gsl_matrix_alloc(p, p);
+
+    gsl_multifit_nlinear_init(_fit_params, &_fdf, work);
+    gsl_blas_ddot(f, f, &_fit_status.chisq0);
+    int status =
+        gsl_multifit_nlinear_driver(_max_iter, xtol, gtol, ftol, print_iter ? Callback : nullptr, nullptr, &_fit_status.info, work);
+    gsl_blas_ddot(f, f, &_fit_status.chisq);
+    gsl_multifit_nlinear_rcond(&_fit_status.rcond, work);
+    gsl_vector_memcpy(_fit_params, y);
+
+    gsl_matrix* jac = gsl_multifit_nlinear_jac(work);
+    gsl_multifit_nlinear_covar(jac, 0.0, covar);
+    const double c = sqrt(_fit_status.chisq / (_fit_data.n - p));
+    for (size_t i = 0; i < p; i++) {
+        gsl_vector_set(_fit_errors, i, c * sqrt(gsl_matrix_get(covar, i, i)));
+    }
+
+    _fit_status.method = fmt::format("{}/{}", gsl_multifit_nlinear_name(work), gsl_multifit_nlinear_trs_name(work));
+    _fit_status.num_iter = gsl_multifit_nlinear_niter(work);
+
+    gsl_multifit_nlinear_free(work);
+    gsl_matrix_free(covar);
+    return status;
+}
+
+void ImageFitter::SetResults() {
+    _results = "";
+    for (size_t i = 0; i < _num_components; i++) {
+        _results += fmt::format("Component #{}:\n", i + 1);
+        _results +=
+            fmt::format("Center X  = {:6f} +/- {:6f} (px)\n", gsl_vector_get(_fit_params, i * 6), gsl_vector_get(_fit_errors, i * 6));
+        _results += fmt::format(
+            "Center Y  = {:6f} +/- {:6f} (px)\n", gsl_vector_get(_fit_params, i * 6 + 1), gsl_vector_get(_fit_errors, i * 6 + 1));
+        _results += fmt::format("Amplitude = {:6f} +/- {:6f}{}\n", gsl_vector_get(_fit_params, i * 6 + 2),
+            gsl_vector_get(_fit_errors, i * 6 + 2), _image_unit.size() > 0 ? fmt::format(" ({})", _image_unit) : "");
+        _results += fmt::format(
+            "FWHM X    = {:6f} +/- {:6f} (px)\n", gsl_vector_get(_fit_params, i * 6 + 3), gsl_vector_get(_fit_errors, i * 6 + 3));
+        _results += fmt::format(
+            "FWHM Y    = {:6f} +/- {:6f} (px)\n", gsl_vector_get(_fit_params, i * 6 + 4), gsl_vector_get(_fit_errors, i * 6 + 4));
+        _results += fmt::format(
+            "P.A.      = {:6f} +/- {:6f} (deg)\n", gsl_vector_get(_fit_params, i * 6 + 5), gsl_vector_get(_fit_errors, i * 6 + 5));
+        _results += "\n";
+    }
+}
```

Comment:
I don't know if this has been discussed before or is planned for a future change, but would it not be cleaner to send raw component data in the response and format it into a string on the client side, instead of sending an arbitrary string formatted on the server side?

We could make a submessage for a value+error tuple (this could have a generic name, since it could potentially be reused elsewhere), and return a map of names:tuples in the response.

Is this too time-consuming to change now?

Reply:
I planned to change this in the next PR because we will need to convert the values into world coordinate. But I think we can make the changes now since it's raised.

Reply:
@confluence @veggiesaurus fixed this issue. also changed `GaussianComponent` to doubles in order to show values with 6 digit precision in frontend.